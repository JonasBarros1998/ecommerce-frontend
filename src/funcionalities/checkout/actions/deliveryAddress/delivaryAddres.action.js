import {
  DELIVERY_ADDRESS,
  NOT_EXIST_ADDRESS,
  CHANGE_ADDRESS
} from '../../constants/checkout.constants'
import { saveLocalStorage, searchLocalStorage } from '../../../../utils/localstorage'

const delivery = (addressDelivery) => {
  return {
    type: DELIVERY_ADDRESS,
    addressDelivery,
  }
}

const searchAddress = () => {
  return {
    type: NOT_EXIST_ADDRESS
  }
}

const updatingAddress = (address) => {
  return {
    type: CHANGE_ADDRESS,
    address
  }
}

/**Função para verificar se existe algum endereço no localstorage. 
 * Se existir carregamos o endereço do usuario.
 * Caso não existir, renderizamos um botão de adicionar endereço. **/
export const verifieldLocalStorage = () => {
  const address = searchLocalStorage('address')
  return (dispatch) => {
    if (address == null) {
      dispatch(searchAddress())
    } else {
      const addressLocalStorage = JSON.parse(address)
      dispatch(delivery(...addressLocalStorage))
    }
  }
}

/** Função usada para o usuario cadastrar um novo endereço **/
export const deliveryAddres = (address) => {
  //salvar as informações do usuario no localstorage
  saveLocalStorage("address", JSON.stringify([address]))
  return (dispatch) => {
    dispatch(delivery(address))
  }
}

/** Função para recuperar os dados do cliente, e trocar de endereço **/
export const changeAddress = () => {
  const addresLocalStorage = searchLocalStorage('address')
  const convertAddress = JSON.parse(addresLocalStorage)
  return dispatch => {
    dispatch(updatingAddress(convertAddress))
  }
} 