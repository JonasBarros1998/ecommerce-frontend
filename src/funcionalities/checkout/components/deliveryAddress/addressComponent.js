import React from 'react'
import { connect } from 'react-redux'

const AddressComponent = props => {

    return (
        <>
        </>
    )
}
/*
const mapStateToProps = store => {
return {
        messageComparationPassword: store.registerUser.messageComparationPassword,
    messageLevelPassword: store.registerUser.messageLevelPassword,
    messageCountPassword: store.registerUser.messageCountPassword
}
}
 
const mapDispatchToProps = dispatch => (bindActionCreators({registerNewUser}, dispatch))
    */

export default connect(null, null)(AddressComponent)