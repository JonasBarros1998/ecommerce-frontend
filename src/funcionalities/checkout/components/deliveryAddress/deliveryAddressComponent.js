/**
 * Componente para rendereizar o endereço de entrega do cliente, e os 
 * itens que ele comprou
 */
import React from 'react'
import OrderItensComponent from './orderItensComponent'
import DeliveryAddressContainer from '../../containers/deliveryAddress/deliveryAddressContainer'
import ModalComponent from '../../../../components/modal/modalComponent'
import ChangeAddressContainer from '../../containers/deliveryAddress/changeAddressContainer'
import SearchAddressContainer from '../../containers/searchAddressContainer'
import { connect } from 'react-redux'

const DeliveryAddressComponent = props => {
    const { loadingAddress } = props
    return (
        <>
            <SearchAddressContainer />
            <div className="head cyan-default mb-1">
                <h5 className="color-white-default bc-cyan pt-2 pb-2 pl-2 mtr-2">Resumo do pedido</h5>
            </div>
            <div className="card billing_details">
                <div className="card-header">Endereço de entrega</div>
                <div className="card-body bc">
                    {
                        /**Se o array de  loadingAddress.length for igual a 0, 
                         * quer dizer que não existe nenhum enderço salvo no 
                         * localStorage. 
                         */
                        loadingAddress.length === 0 ?
                            <div className="d-flex justify-content-center">
                                <button type="button"
                                    className="genric-btn success text-uppercase"
                                    data-toggle="modal"
                                    data-target="#modal_address">Adicionar endereço</button>
                            </div>

                            /** Renderizamos o endereço se o array de loadingAddress
                             * não estiver vazio, ou seja, já ha um endereço salvo no localStorage*/
                            : props.loadingAddress.map((item, index) => (
                                <p className="color-grimmys-grey" key={index}>
                                    {item.street}<br />
                                    {item.burgh}, {item.number}<br />
                                    São Paulo- Sp<br />
                                    {item.cep}</p>
                            ))
                    }

                    {/*Se o endereço existir, vai retornar true, e renderizar 
                            * o botão  trocar de endereço*/
                        props.existAddress === true ?
                            <>
                                <hr />
                                <div>
                                    <small className="color-grimmys-grey">Quer receber seus produtos em outro endereço?</small>
                                </div>
                                {/** Container que renderiza o botão para trocar de endereço **/}
                                <ChangeAddressContainer />
                            </>
                            : <div></div>
                    }
                </div>
            </div>

            {/** Componente para renderizar os produtos 
             * comprados pelo cliente**/}
            <OrderItensComponent />

            {
                /** Modal para alteração de endereço
                 * Só renderizamos o ModalComponent se o endereço do cliente já estiver
                 * salvo no localStorage*/

                //Se loadingAddress.length for diferente de 0, renderizamos a modal
                loadingAddress.length !== 0 ?
                    <ModalComponent
                        id={"modal_center"}
                        ariaLabelledby={"myLargeModalLabel"}
                        classModal={"modal-dialog modal-lg"}
                        title={<DeliveryAddressContainer
                            id={"modal_center"}
                            nameFields={loadingAddress} />}
                        footer={false} />
                    //Se for igual a zero, renderizamos uma div vazia
                    : <div></div>
            }

            {
                /** Modal para criação de um novo endereço
                 * Só renderizamos esse modal se não existir algum endereço cadastrado pelo cliente
                 */

                <ModalComponent
                    id={"modal_address"}
                    ariaLabelledby={"myLargeModalLabel"}
                    classModal={"modal-dialog modal-lg"}
                    title={<DeliveryAddressContainer
                        id="modal_address"
                        nameFields={[{
                            email: "",
                            address: "",
                            toReceiver: "",
                            cep: "",
                            typeAddress: "",
                            street: "",
                            number: "",
                            complement: "",
                            burgh: "",
                            states: "",
                            city: "",
                            reference: "",
                            phone: ""
                        }]}
                    />}
                    body={"Cadastre seu endereço"}
                    footer={false} />
            }
        </>
    )
}

const mapStateToProps = store => {
    return {
        loadingAddress: store.checkout.loadingAddress,
        existAddress: store.checkout.existAddress,
    }
}

export default connect(mapStateToProps)(DeliveryAddressComponent) 