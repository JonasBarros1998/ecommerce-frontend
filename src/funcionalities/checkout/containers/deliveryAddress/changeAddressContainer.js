/**
 * container especifico para renderizar o botão de atuallizar o endereço de entrega do cliente
 */
import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { changeAddress } from '../../actions/deliveryAddress/delivaryAddres.action'

const ChangeAddressContainer = props => {

    return (
        <>
            {
                <button type="button"
                    className="genric-btn success text-uppercase"
                    data-toggle="modal"
                    data-target="#modal_center"
                    onClick={() => props.changeAddress()}>Alterar endereço</button>
            }
        </>
    )
}

const mapDispatchToProps = dispatch => bindActionCreators({ changeAddress }, dispatch)
export default connect(null, mapDispatchToProps)(ChangeAddressContainer)
