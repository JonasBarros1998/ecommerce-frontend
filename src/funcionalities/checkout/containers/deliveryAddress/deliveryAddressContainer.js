/**
 * Componente que tem como funcionalidade renderizar 
 * o formulario para cadastro e edição do endereço do cliente. 
 */
import React, { useState } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import InputMask from 'react-input-mask'
import $ from 'jquery'
import SelectComponent from '../../../../components/select/selectComponent'
import { deliveryAddres } from '../../actions/deliveryAddress/delivaryAddres.action'

export const DeliveryAddressContainer = props => {

    const { nameFields } = props

    const [changeAddress] = useState(...nameFields)
    const [textContentSelect, setTextContentSelect] = useState("")

    const [form, setForm] = useState({
        email: changeAddress.email,
        address: changeAddress.address,
        toReceiver: changeAddress.toReceiver,
        cpf: changeAddress.cpf,
        cep: changeAddress.cep,
        typeAddress: changeAddress.typeAddress,
        street: changeAddress.street,
        number: changeAddress.number,
        complement: changeAddress.complement,
        burgh: changeAddress.burgh,
        states: changeAddress.states,
        city: changeAddress.city,
        reference: changeAddress.reference,
        phone: changeAddress.phone,
    })

    //Atualizar o state do select
    const updateSelect = (event) => {
        const valueSelect = event.target.value
        const selectId = event.target.id
        const select = document.querySelector(`#${selectId}`)
        const selectTextContent = select[event.target.value].textContent
        setTextContentSelect(selectTextContent)
        return setForm({
            ...form,
            [selectId]: valueSelect
        })
    }
    //Atualizar o state dos inputs
    const upadateInput = (event) => {
        return setForm({
            ...form,
            [event.target.name]: event.target.value
        })
    }

    const { email, address, toReceiver,
        cpf, cep, typeAddress, street, number,
        complement, burgh, states,
        city, reference, phone } = form

    return (
        <>
            <form className="row contact_form" method="post"
                onSubmit={(e) => {
                    e.preventDefault();
                    props.deliveryAddres({
                        email: email, address: address, toReceiver: toReceiver, cpf: cpf,
                        cep: cep, typeAddress: typeAddress, street: street, number: number,
                        complement: complement, burgh: burgh, 
                        states: {stateValue: states, nameState: textContentSelect}, city: city,
                        reference: reference, phone: phone
                    })
                }}>

                <div className="row ml-2">
                    <div className="col-md-12 form-group">
                        <label><small>e-mail</small></label>
                        <input type='email' className='form-control' id='registerEmail'
                            name='email' placeholder='exemplo@exemplo.com'
                            value={email}
                            onChange={e => upadateInput(e)} />
                    </div>

                    <div className="col-md-12 form-group">
                        <label> <small>Nome do endereço</small></label>
                        <input type='text' className='form-control'
                            id='nameAddress' name='address'
                            placeholder="Ex: Minha casa, Meu trabalho"
                            value={address}
                            onChange={e => upadateInput(e)} />
                    </div>

                    <div className="col-md-9 form-group">
                        <label><small>Nome do destinatário</small></label>
                        <input type="text" className='form-control' id="nameRecipient"
                            name="toReceiver" placeholder='Nome de quem vai receber a encomenda'
                            value={toReceiver}
                            onChange={e => upadateInput(e)} />
                    </div>

                    <div className="col-md-3 form-group">
                        <label><small>Cpf</small></label>
                        <InputMask type="text" className="form-control" id="cpf"
                            name="cpf" placeholder="000.000.000.00" mask="999.999.999-99"
                            value={cpf}
                            onChange={e => upadateInput(e)}
                            required />
                    </div>

                    <div className="col-md-3 form-group">
                        <label><small>Cep</small></label>
                        <InputMask type="text" className="form-control" id="cep"
                            name="cep" placeholder="00.000-000" mask="99.999-999"
                            value={cep}
                            onChange={e => upadateInput(e)}
                            required />
                    </div>

                    <div className="col-md-9 form-group">
                        <label> <small>Nome da rua</small></label>
                        <input type='text' className='form-control' id='nameStreet'
                            name='street' placeholder="Nome da rua"
                            value={street}
                            onChange={e => upadateInput(e)} required />
                    </div>

                    <div className="col-md-2 form-group">
                        <label><small>Número</small></label>
                        <input type="text" className="form-control" id="number"
                            name="number" placeholder="Número da residência"
                            value={number}
                            onChange={e => upadateInput(e)} required />
                    </div>

                    <div className="col-md-6 form-group">
                        <label><small>Complemento</small></label>
                        <input type="text" className="form-control" id="complement"
                            name="complement" placeholder="*Ex: Apartamento 51 5 andar, bloco B"
                            value={complement}
                            onChange={e => upadateInput(e)} />
                    </div>

                    <div className="col-md-4 form-group">
                        <label><small>Bairro</small></label>
                        <input type="text" className="form-control" id="burgh" name="burgh"
                            value={burgh} placeholder="Nome do bairro de entrega"
                            onChange={e => upadateInput(e)}
                            required />
                    </div>

                    <div className="col-md-4 form-group">
                        <label><small>Cidade</small></label>
                        <input className="form-control" id="city" name="city"
                            value={city}
                            onChange={e => upadateInput(e)} />
                    </div>

                    <div className="col-md-4 form-group">
                        <label>
                            <small>Estado</small>
                        </label>
                        <SelectComponent id={"states"}
                            class={"custom-select mr-sm-2 form-control select-delivery"}
                            select={e => updateSelect(e)}
                            value={states.stateValue} />
                    </div>

                    <div className="col-md-4 form-group">
                        <label><small>Ponto de referência</small></label>
                        <input type="text" className="form-control" id="referencePoint"
                            name="reference"
                            value={reference}
                            onChange={e => upadateInput(e)}
                            required />
                    </div>

                    <div className="col-md-12 form-group">
                        <label><small>Telefone</small></label>
                        <InputMask type="text" className="form-control" id="residentialPhone"
                            mask="(99) 9999-99999" name="phone"
                            placeholder="(00) 0000-00000"
                            value={phone}
                            onChange={e => upadateInput(e)}
                            required />
                    </div>

                    <div className="col-md-12">
                        <div className="d-flex justify-content-end">
                            <button type="submit"
                                value="submit"
                                className="genric-btn success mt-2"
                                onClick={() => $(`#${props.id}`).modal('hide')}>Salvar</button>
                        </div>
                    </div>

                </div>
            </form>
        </>
    )
}

const mapStateToProps = (store) => ({ loadingAddress: store.checkout.loadingAddress })
const mapDispatchToProps = (dispatch) => bindActionCreators({ deliveryAddres }, dispatch)
export default connect(mapStateToProps, mapDispatchToProps)(DeliveryAddressContainer)