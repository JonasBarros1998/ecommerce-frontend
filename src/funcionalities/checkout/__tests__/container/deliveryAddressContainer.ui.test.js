/**
 * Teste de UI(user interface) do componente de formulario de 
 * cadastro de endereço
 */
import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { DeliveryAddressContainer } from '../../containers/deliveryAddress/deliveryAddressContainer'

Enzyme.configure({ adapter: new Adapter() })

const address = [{
    email: "jonas_barros@outlook.com",
    address: "Minha casa",
    toReceiver: "Jonas Florencio de Barros",
    cep: "04851410",
    typeAddress: "Avenida",
    street: "Rua andorinhas brasileiras",
    number: "61",
    complement: "Casa",
    burgh: "Lago azul",
    states: "São Paulo",
    city: "Sp",
    reference: "Frente a igreja católica",
    phone: "11984308992"
}]

const setup = () => {
    const props = {
        addTodo: jest.fn()
    }
    const deliveryAddressContainer = shallow(<DeliveryAddressContainer nameFields={address} />)

    return {
        props,
        deliveryAddressContainer
    }
}

describe('Component de cadastro de endereço de entrega de produtos', () => {
    const { deliveryAddressContainer } = setup()
     it("Teste de snaptshot do componente de endereço", () => {
         expect(deliveryAddressContainer.html()).toMatchSnapshot();
     });

    it("Campo de e-mail", () => {
        expect(deliveryAddressContainer.find('input[name="email"]').props()).toEqual({
            className: 'form-control', id: "registerEmail", name: "email", type: "email",
            onChange: expect.any(Function),
            placeholder: "exemplo@exemplo.com",
            value: "jonas_barros@outlook.com"
        })
    })

    it("campo de nome do endereço", () => {
        expect(deliveryAddressContainer.find('input[name="nameAddress"]').props()).toEqual({
            className: 'form-control', id: "nameAddress", name: "nameAddress", type: "text",
            onChange: expect.any(Function),
            placeholder: "Ex: Minha casa, Meu trabalho",
            value: "Minha casa",
        })
    })

    it("campo de nome do destinatario", () => {
        expect(deliveryAddressContainer.find('input[name="nameRecipient"]').props()).toEqual({
            className: "form-control", id: "nameRecipient", name: "nameRecipient", type: "text",
            onChange: expect.any(Function),
            placeholder: "Nome de quem vai receber a encomenda",
            value: "Jonas Florencio de Barros"
        })
    })

    it("campo de nome do destinatario", () => {
        expect(deliveryAddressContainer.find('#cep').props()).toEqual({
            className: "form-control", id: "cep", name: "cep", type: "text",
            required: true,
            mask: "99.999-999",
            onChange: expect.any(Function),
            placeholder: "00.000-000",
            value: "04851410"
        })
    })

    it("campo de tipo de endereco", () => {
        expect(deliveryAddressContainer.find('#typeAddress').props()).toEqual({
            class: "custom-select mr-sm-2 form-control select-delivery",
            id: "typeAddress",
            select: expect.any(Function)
        })
    })

    it("campo de cep", () => {
        expect(deliveryAddressContainer.find('input[name="nameStreet"]').props()).toEqual({
            className: "form-control", id: "nameStreet", name: "nameStreet", type: "text",
            required: true,
            onChange: expect.any(Function),
            placeholder: "Nome da rua",
            value: "Rua andorinhas brasileiras"
        })
    })

    it("campo de numero da residencia", () => {
        expect(deliveryAddressContainer.find('input[name="number"]').props()).toEqual({
            className: "form-control", id: "number", name: "number", type: "text",
            required: true,
            onChange: expect.any(Function),
            placeholder: "Número da residência",
            value: "61"
        })
    })

    it("campo de complemento", () => {
        expect(deliveryAddressContainer.find('input[name="complement"]').props()).toEqual({
            className: "form-control", id: "complement", name: "complement", type: "text",
            onChange: expect.any(Function),
            placeholder: "*Ex: Apartamento 51 5 andar, bloco B",
            value: "Casa"
        })
    })

    it("campo de complemento", () => {
        expect(deliveryAddressContainer.find('input[name="burgh"]').props()).toEqual({
            className: "form-control", id: "burgh", name: "burgh", type: "text",
            required: true,
            onChange: expect.any(Function),
            placeholder: "Nome do bairro de entrega",
            value: "Lago azul"
        })
    })

    it("campo de estados", () => {
        expect(deliveryAddressContainer.find('#states').props()).toEqual({
            class: "custom-select mr-sm-2 form-control select-delivery",
            id: "states",
            select: expect.any(Function)
        })
    })

    it("campo de ponto de referencia", () => {
        expect(deliveryAddressContainer.find('input[name="referencePoint"]').props()).toEqual({
            className: "form-control", id: "referencePoint", name: "referencePoint", type: "text",
            required: true,
            onChange: expect.any(Function),
            value: "Frente a igreja católica"
        })
    })

    it("campo de telefone", () => {
        expect(deliveryAddressContainer.find('#residentialPhone').props()).toEqual({
            className: "form-control", id: "residentialPhone", name: "residentialPhone", type: "text",
            mask: "(99) 9999-99999",
            required: true,
            onChange: expect.any(Function),
            placeholder: "(00) 0000-00000",
            value: "11984308992"
        })
    })

})


