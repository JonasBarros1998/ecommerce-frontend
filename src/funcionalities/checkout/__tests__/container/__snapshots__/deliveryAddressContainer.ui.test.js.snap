// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component de cadastro de endereço de entrega de produtos Teste de snaptshot do componente de endereço 1`] = `"<form class=\\"row contact_form\\" method=\\"post\\"><div class=\\"row ml-2\\"><div class=\\"col-md-12 form-group\\"><label><small>e-mail</small></label><input type=\\"email\\" class=\\"form-control\\" id=\\"registerEmail\\" name=\\"email\\" placeholder=\\"exemplo@exemplo.com\\" value=\\"jonas_barros@outlook.com\\"/></div><div class=\\"col-md-12 form-group\\"><label> <small>Nome do endereço</small></label><input type=\\"text\\" class=\\"form-control\\" id=\\"nameAddress\\" name=\\"nameAddress\\" placeholder=\\"Ex: Minha casa, Meu trabalho\\" value=\\"Minha casa\\"/></div><div class=\\"col-md-12 form-group\\"><label><small>Nome do destinatário</small></label><input type=\\"text\\" class=\\"form-control\\" id=\\"nameRecipient\\" name=\\"nameRecipient\\" placeholder=\\"Nome de quem vai receber a encomenda\\" value=\\"Jonas Florencio de Barros\\"/></div><div class=\\"col-md-3 form-group\\"><label><small>Cep</small></label><input type=\\"text\\" class=\\"form-control\\" id=\\"cep\\" name=\\"cep\\" placeholder=\\"00.000-000\\" value=\\"04.851-410\\" required=\\"\\"/></div><div class=\\"col-md-3 align-items-center\\"><label><small>Tipo de endereço</small></label><select class=\\"custom-select mr-sm-2 form-control select-delivery\\" id=\\"typeAddress\\"><option>Selecione...</option><option value=\\"1\\">Avenida</option><option value=\\"2\\">Rua</option><option value=\\"3\\">Praça</option><option value=\\"4\\">Quadra</option><option value=\\"5\\">Estrada</option><option value=\\"6\\">Alameda</option><option value=\\"7\\">Ladeira</option><option value=\\"8\\">Travessa</option><option value=\\"9\\">Rodovia</option></select></div><div class=\\"col-md-6 form-group\\"><label> <small>Nome da rua</small></label><input type=\\"text\\" class=\\"form-control\\" id=\\"nameStreet\\" name=\\"nameStreet\\" placeholder=\\"Nome da rua\\" value=\\"Rua andorinhas brasileiras\\" required=\\"\\"/></div><div class=\\"col-md-2 form-group\\"><label><small>Número</small></label><input type=\\"text\\" class=\\"form-control\\" id=\\"number\\" name=\\"number\\" placeholder=\\"Número da residência\\" value=\\"61\\" required=\\"\\"/></div><div class=\\"col-md-6 form-group\\"><label><small>Complemento</small></label><input type=\\"text\\" class=\\"form-control\\" id=\\"complement\\" name=\\"complement\\" placeholder=\\"*Ex: Apartamento 51 5 andar, bloco B\\" value=\\"Casa\\"/></div><div class=\\"col-md-4 form-group\\"><label><small>Bairro</small></label><input type=\\"text\\" class=\\"form-control\\" id=\\"burgh\\" name=\\"burgh\\" value=\\"Lago azul\\" placeholder=\\"Nome do bairro de entrega\\" required=\\"\\"/></div><div class=\\"col-md-4 form-group\\"><label><small>Estado</small></label><select class=\\"custom-select mr-sm-2 form-control select-delivery\\" id=\\"states\\"><option>Selecione...</option><option value=\\"1\\">Acre</option><option value=\\"2\\">Alagoas</option><option value=\\"3\\">Amapá</option><option value=\\"4\\">Amazonas</option><option value=\\"5\\">Bahia</option><option value=\\"6\\">Ceará</option><option value=\\"7\\">Distrito Federal</option><option value=\\"8\\">Espírito Santo</option><option value=\\"9\\">Goiás</option><option value=\\"10\\">Maranhão</option><option value=\\"11\\">Mato Grosso</option><option value=\\"12\\">Mato Grosso do Sul</option><option value=\\"13\\">Minas Gerais</option><option value=\\"14\\">Pará</option><option value=\\"15\\">Paraíba</option><option value=\\"16\\">Paraná</option><option value=\\"17\\">Pernambuco</option><option value=\\"18\\">Piauí</option><option value=\\"19\\">Rio de Janeiro</option><option value=\\"20\\">Rio Grnade do Norte</option><option value=\\"21\\">Rio Grande do Sul</option><option value=\\"22\\">Rondônia</option><option value=\\"23\\">Roraima</option><option value=\\"24\\">Santa Catarina</option><option value=\\"25\\">São Paulo</option><option value=\\"26\\">Sergipe</option><option value=\\"27\\">Tocantins</option></select></div><div class=\\"col-md-4 form-group\\"><label><small>Cidade</small></label><input class=\\"form-control\\" id=\\"city\\" name=\\"city\\" value=\\"Sp\\"/></div><div class=\\"col-md-4 form-group\\"><label><small>Ponto de referência</small></label><input type=\\"text\\" class=\\"form-control\\" id=\\"referencePoint\\" name=\\"referencePoint\\" value=\\"Frente a igreja católica\\" required=\\"\\"/></div><div class=\\"col-md-12 form-group\\"><label><small>Telefone</small></label><input type=\\"text\\" class=\\"form-control\\" id=\\"residentialPhone\\" name=\\"residentialPhone\\" placeholder=\\"(00) 0000-00000\\" value=\\"(11) 9843-08992\\" required=\\"\\"/></div><div class=\\"col-md-12\\"><div class=\\"d-flex justify-content-end\\"><button type=\\"submit\\" value=\\"submit\\" class=\\"genric-btn success mt-2\\">Prosseguir</button></div></div></div></form>"`;
